woocommerce/woocommerce-android
# [Product AI v2] Tracking + enabling feature

Closes: #12054 Closes: #12063


⚠️ Please don't merge this, I'll handle merging myself.

### Description
This PR adds the tracking to the Product Creation with AI.

### Testing information
1. Use a Jetpack website with Jetpack AI (atomic websites will work fine)
2. Monitor `Logcat`
3. Open the app, then open the product list.
4. Tap on Add button.
5. Confirm the event `Tracked: product_creation_ai_entry_point_displayed`
6. Tap on "Create a product with AI"
7. Confirm the event `Tracked: product_creation_ai_entry_point_tapped`
8. Tap on "Read text from product photo"
9. Confirm the event `Tracked: product_creation_ai_started_package_photo_selection_flow`
10. Select a photo that has some text.
11. Confirm the event `Tracked: product_creation_ai_text_detected, Properties: {"number_of_texts": (number of words detected)}`
12. Tap on "Generate Product Details"
13. Confirm the event `Tracked: product_creation_ai_generate_details_tapped, Properties: {"is_first_attempt":true}`
14. Go back to the previous screen, then tap again on "Generate Product Details"
15. Confirm the event `Tracked: product_creation_ai_generate_details_tapped, Properties: {"is_first_attempt":**false**}`
16. Wait for the product generation to finish.
17. Confirm the events:
`Tracked: product_creation_ai_generate_product_details_success`
`Tracked: product_creation_ai_generated_name_description_options, Properties: {"name":3, "short_description":3, "description":3}`
18. Edit one of the text fields (name, summary or description)
19. Tap on the Undo button.
20. Confirm the event `Tracked: product_creation_ai_undo_edit_tapped, Properties: {"field":"(name of the field)"}`
21. Tap on "Save as draft"
22. Confirm the event `Tracked: product_creation_ai_save_as_draft_button_tapped`
23. When the generation succeeds, confirm the event `Tracked: product_creation_ai_save_as_draft_success`
